integration-tests/tests/src/tests/sync/mixed.ts

```diff
@@ -123,8 +123,16 @@ function describeRoundtrip({
     it("reads", async function (this: RealmContext) {
       await setupTest(this.realm);
 
-      const obj = this.realm.objectForPrimaryKey<MixedClass>("MixedClass", this._id);
-      if (!obj) throw new Error("Object not found");
+      const obj = await new Promise<MixedClass>((resolve) => {
```

Comment:
It can wait until we need it, but a generic method `waitForObject<T>(realm, id): Promise<T>` could be useful.